tests/cases/csv_points_source.py

```diff
@@ -0,0 +1,125 @@
+import random
+
+import numpy as np
+import pytest
+import csv
+
+from gunpowder import (
+    BatchRequest,
+    CsvPointsSource,
+    GraphKey,
+    GraphSpec,
+    build,
+    Coordinate,
+    Roi,
+)
+
+
+# automatically set the seed for all tests
+@pytest.fixture(autouse=True)
+def seeds():
+    random.seed(12345)
+    np.random.seed(12345)
+
+
+@pytest.fixture
+def test_points_2d(tmpdir):
+    random.seed(1234)
+    np.random.seed(1234)
+
+    fake_points_file = tmpdir / "shift_test.csv"
+    fake_points = np.random.randint(0, 100, size=(2, 2))
+    with open(fake_points_file, "w") as f:
+        for point in fake_points:
+            f.write(str(point[0]) + "\t" + str(point[1]) + "\n")
+
+    # This fixture will run after seeds since it is set
+    # with autouse=True. So make sure to reset the seeds properly at the end
+    # of this fixture
+    random.seed(12345)
+    np.random.seed(12345)
+
+    yield fake_points_file, fake_points
+
+
+@pytest.fixture
+def test_points_3d(tmpdir):
+    random.seed(1234)
+    np.random.seed(1234)
+
+    fake_points_file = tmpdir / "shift_test.csv"
+    fake_points = np.random.randint(0, 100, size=(3, 3)).astype(float)
+    with open(fake_points_file, "w") as f:
+        writer = csv.DictWriter(f, fieldnames=["x", "y", "z", "id"])
+        writer.writeheader()
+        for i, point in enumerate(fake_points):
+            pointdict = {"x": point[0], "y": point[1], "z": point[2], "id": i}
+            writer.writerow(pointdict)
+
+    # This fixture will run after seeds since it is set
+    # with autouse=True. So make sure to reset the seeds properly at the end
+    # of this fixture
+    random.seed(12345)
```

Comment:
The seed setting seems a bit overkill. Looks like the only randomness is in the generation of the `fake_points`. So since the `autouse=True` seed fixture will run before either of the other two fixtures you can remove all of the seed setting in the other fixtures.